### PR TITLE
[docs] unet type hints

### PR DESCRIPTION
**Pull request by a-r-r-o-w**
# What does this PR do?

Addresses https://github.com/huggingface/diffusers/pull/7044#discussion_r1506473969 along with some other minor type annotation changes.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@yiyixuxu

### CODE REVIEW

File: src/diffusers/models/unets/unet_2d_condition.py [hunk 1]
```diff
@@ -762,7 +762,7 @@ def set_default_attn_processor(self):
 
         self.set_attn_processor(processor)
 
-    def set_attention_slice(self, slice_size):
+    def set_attention_slice(self, slice_size: Union[str, int, List[int]] = "auto"):
```
Comment (a-r-r-o-w) [inline src/diffusers/models/unets/unet_2d_condition.py:765]: this has been done to match the docs @yiyixuxu. if we don't want it to be `auto` by default, i can update the docs accordingly

Comment (yiyixuxu) [reply src/diffusers/models/unets/unet_2d_condition.py:765]: I think it's ok

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Review by stevhliu — APPROVED** [review]
Nice, thanks for adding!

**Pull request merged**